pkg/estimator/client/cache.go

```diff
@@ -98,9 +89,22 @@ func (c *SchedulerEstimatorCache) GetClient(name string) (estimatorservice.Estim
 
 // EstablishConnection establishes a new gRPC connection with the specified cluster scheduler estimator.
 func EstablishConnection(kubeClient kubernetes.Interface, name string, estimatorCache *SchedulerEstimatorCache, estimatorServicePrefix string, grpcConfig *grpcconnection.ClientConfig) error {
-	if estimatorCache.IsEstimatorExist(name) {
-		return nil
+	// check if the connection already exists, if so, close it and establish a new connection
+	estimatorCache.lock.Lock()
+	es, exist := estimatorCache.estimator[name]
+	if exist {
+		// Use a lock to prevent if more than one worker close connection at the same time,
+		// a connection may be closed multiple times.
```

Comment:
is it possible that multiple workers can acquire this lock `estimatorCache.lock` at the same time?

Reply:
BTW, I think this close function `es.connection.Close()` is thread safe.

Reply:
> I think this close function es.connection.Close() is thread safe.

It is indeed thread-safe, but it may cause some unused connections to not be closed properly.

Reply:
> is it possible that multiple workers can acquire this lock `estimatorCache.lock` at the same time?

`EstablishConnection` is uesd as a part of `ReconcileFunc` of scheduler/descheduler.schedulerEstimatorWorker, and has the ability to run concurrently.